feat: add `blas/ext/base/gfill-nan` (#13272)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/gfill-nan/README.md

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# gfill-nan

> Replace strided array elements equal to `NaN` with a specified scalar constant.

<section class="usage">

## Usage

```javascript
var gfillNaN = require( '@stdlib/blas/ext/base/gfill-nan' );
```

#### gfillNaN( N, alpha, x, strideX )

Replaces strided array elements equal to `NaN` with a specified scalar constant.

```javascript
var x = [ NaN, 1.0, 3.0, NaN, 4.0, 0.0, -1.0, -3.0 ];

gfillNaN( x.length, 0.0, x, 1 );
// x => [ 0.0, 1.0, 3.0, 0.0, 4.0, 0.0, -1.0, -3.0 ]
```

The function has the following parameters:

-   **N**: number of indexed elements.
-   **alpha**: scalar constant.
-   **x**: input array.
-   **strideX**: stride length.

> The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to replace every other NaN element:

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to iterate over every other element:
```

```javascript
var x = [ NaN, 1.0, NaN, -5.0, NaN, 0.0, -1.0, -3.0 ];

gfillNaN( 4, 0.0, x, 2 );
// x => [ 0.0, 1.0, 0.0, -5.0, 0.0, 0.0, -1.0, -3.0 ]
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

// Initial array...
var x0 = new Float64Array( [ 1.0, NaN, 3.0, NaN, 5.0, -6.0 ] );

> // Create an offset view...

Reviewer: ```suggestion
// Initial array:
var x0 = new Float64Array( [ 1.0, NaN, 3.0, NaN, 5.0, -6.0 ] );

// Create an offset view:
```

var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

> // Iterate every other element:

Reviewer: ```suggestion
// Iterate over every other element:
```

> gfillNaN( 3, 0.0, x1, 2 );

Reviewer: ```suggestion
// Iterate every other element:
gfillNaN( 3, 0.0, x1, 2 );
```

<!--

@license Apache-2.0

Copyright (c) 2026 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# gfill-nan

> > Replace strided array elements equal to NaN with a specified scalar constant.

Reviewer: ```suggestion
> Replace strided array elements equal to `NaN` with a specified scalar constant.
```

<section class="usage">

## Usage

```javascript
var gfillNaN = require( '@stdlib/blas/ext/base/gfill-nan' );
```

#### gfillNaN( N, alpha, x, strideX )

> Replaces strided array elements equal to NaN with a specified scalar constant.

Reviewer: ```suggestion
Replaces strided array elements equal to `NaN` with a specified scalar constant.
```

var x = [ NaN, 1.0, 3.0, NaN, 4.0, 0.0, -1.0, -3.0 ];

gfillNaN( x.length, 0.0, x, 1 );
// x => [ 0.0, 1.0, 3.0, 0.0, 4.0, 0.0, -1.0, -3.0 ]
```

The function has the following parameters:

-   **N**: number of indexed elements.
-   **alpha**: scalar constant.
-   **x**: input array.
-   **strideX**: stride length.

The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to replace every other NaN element:

```javascript
var x = [ NaN, 1.0, NaN, -5.0, NaN, 0.0, -1.0, -3.0 ];

gfillNaN( 4, 0.0, x, 2 );
// x => [ 0.0, 1.0, 0.0, -5.0, 0.0, 0.0, -1.0, -3.0 ]
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

// Initial array...
var x0 = new Float64Array( [ 1.0, NaN, 3.0, NaN, 5.0, -6.0 ] );

// Create an offset view...
var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

// Replace every other NaN element...
gfillNaN( 3, 0.0, x1, 2 );
// x0 => <Float64Array>[ 1.0, 0.0, 3.0, 0.0, 5.0, -6.0 ]
```

#### gfillNaN.ndarray( N, alpha, x, strideX, offsetX )

> Replaces strided array elements equal to NaN with a specified scalar constant using alternative indexing semantics.

Reviewer: ```suggestion
Replaces strided array elements equal to `NaN` with a specified scalar constant using alternative indexing semantics.
```

Literal values should be enclosed in backticks.

<section class="notes">

## Notes

-   If `N <= 0`, both functions return `x` unchanged.
-   Both functions support array-like objects having getter and setter accessors for array element access (e.g., [`@stdlib/array/base/accessor`][@stdlib/array/base/accessor]).

</section>

<!-- /.notes -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var nans = require( '@stdlib/array/nans' );
var gfillNaN = require( '@stdlib/blas/ext/base/gfill-nan' );

var x = nans( 10 );
console.log( x );

gfillNaN( x.length, 0.0, x, 1 );
console.log( x );
```

</section>

<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[mdn-typed-array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray

[@stdlib/array/base/accessor]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/accessor

> 

Reviewer: ```suggestion
```
